### PR TITLE
Refactor to Customer model

### PR DESCRIPTION
**Pull request by driesvints**
This PR introduces a new `Customer` model which will allow for much greater flexibility to use different kind of models for your Billing setup. Inspired by https://github.com/laravel/cashier/issues/545

In terms of public API changes not much has changed. `$user->subscriptions` would now become `$user->subscription()` since you're not directly referencing a relationship anymore. There are however, some major differences on the database scheme side with a new customers table which now contains all the Paddle fields.

### Todo

- [x] Make `Customer` model polymorphic
- [x] Proxy method calls on Billable trait to `Customer` model

### CODE REVIEW

File: src/Concerns/ManagesSubscriptions.php [hunk 1]
```diff
@@ -20,13 +19,17 @@ public function newSubscription($name, $plan)
...
+     * Get all of the subscriptions for the Billable model.
      *
-     * @return \Illuminate\Database\Eloquent\Relations\HasMany
+     * @return \Laravel\Paddle\Subscription[]|\Illuminate\Database\Eloquent\Collection
      */
     public function subscriptions()
     {
-        return $this->hasMany(Subscription::class, $this->getForeignKey())->orderByDesc('created_at');
+        if (is_null($this->customer)) {
+            return collect();
```
Comment (mateusjatenee) [inline src/Concerns/ManagesSubscriptions.php:29]: Might be a dumb question, but if the user is doing any operations that are exclusive to Eloquent Collection, wouldn't this be an issue?

Comment (driesvints) [reply src/Concerns/ManagesSubscriptions.php:29]: Not a dumb question at all. I'll update this tomorrow 👍

File: database/migrations/2019_05_03_000001_create_customers_table.php [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+     * @return void
+     */
+    public function up()
+    {
+        Schema::create('customers', function (Blueprint $table) {
+            $table->id();
+            $table->unsignedBigInteger('billable_id');
+            $table->string('billable_type');
+            $table->string('paddle_id')->unique();
+            $table->string('paddle_email')->unique();
```
Comment (stefanzweifel) [inline database/migrations/2019_05_03_000001_create_customers_table.php:21]: Wouldn't it make sense to make both of the columns nullable?

In it's current form it's not possible to create a customer without paddle_id/email. For example in the case of setting the `trial_ends_at` during the default Laravel registration process.

Comment (driesvints) [reply database/migrations/2019_05_03_000001_create_customers_table.php:21]: That's an excellent point. I'll update that 👍

**Pull request merged**